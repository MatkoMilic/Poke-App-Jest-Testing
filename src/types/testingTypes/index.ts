export type {ReactWrapperType, RenderApiType} from './testingTypes';
