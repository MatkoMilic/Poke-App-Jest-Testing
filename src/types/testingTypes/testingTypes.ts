import {RenderAPI} from '@testing-library/react-native';
import {ReactWrapper} from 'enzyme';

export type ReactWrapperType = ReactWrapper<
  any,
  Readonly<{}>,
  React.Component<{}, {}, any>
>;
export type RenderApiType = RenderAPI;
