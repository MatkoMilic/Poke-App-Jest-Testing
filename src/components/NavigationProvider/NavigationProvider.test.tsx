import 'jsdom-global/register';
import React from 'react';
import {act, render} from '@testing-library/react-native';
import {NavigationContainer} from '@react-navigation/native';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ReactWrapperType} from '../../types/testingTypes/testingTypes';
import {NavigationProvider} from './NavigationProvider';
configure({adapter: new Adapter()});

describe('ThemeProvider', () => {
  let mountNavigationProvider: ReactWrapperType;
  let renderNavigationContainer: any;
  let children: React.ReactNode;
  beforeEach(() => {
    mountNavigationProvider = mount(<NavigationProvider />);
    renderNavigationContainer = render(
      <NavigationContainer>{children}</NavigationContainer>,
    ).toJSON();
  });
  it('should match snapshot', async () => {
    await act(async () => {
      expect(renderNavigationContainer).toMatchSnapshot();
    });
  });
});
