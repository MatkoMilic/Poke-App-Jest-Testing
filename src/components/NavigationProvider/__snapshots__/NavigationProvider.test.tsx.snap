// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should match snapshot 1`] = `null`;
